agent/06_agent_create_cluster.sh

```diff
@@ -163,13 +163,74 @@ function mce_complete_deployment() {
   mce_apply_postinstallation_manifests ${mceManifests}
 }
 
-create_image
-if [[ "${AGENT_DISABLE_AUTOMATED:-}" == "true" ]]; then
-  disable_automated_installation
-fi
+# Generate the PXE artifacts
+function create_pxe_files() {
+    local asset_dir=${1}
+    local openshift_install=${2}
+    "${openshift_install}" --dir="${asset_dir}" --log-level=debug agent create pxe-files
+}
+
+# Setup the environment to allow iPXE booting, by reusing libvirt native features
+# to configure dnsmaq tftp server and pxe boot file. Note that this is supported
+# only for IPv4.
+function setup_pxe_server() {
+    local PXE_SERVER_DIR=${WORKING_DIR}/pxe
+    mkdir -p ${PXE_SERVER_DIR}
+
+    # Create the iPXE boot file
+    local BASEURL=tftp://${PROVISIONING_HOST_EXTERNAL_IP}
+    sudo cat > ${PXE_SERVER_DIR}/boot.ipxe << EOF
+#!ipxe
+kernel ${BASEURL}/agent-vmlinuz.x86_64 initrd=main coreos.live.rootfs_url=${BASEURL}/agent-rootfs.x86_64.img ignition.firstboot ignition.platform.id=metal
+initrd --name main ${BASEURL}/agent-initrd.x86_64.img
+boot
+EOF
+
+    # Configure the baremetal network for PXE 
+    sudo virsh net-dumpxml ${BAREMETAL_NETWORK_NAME} > ${WORKING_DIR}/${BAREMETAL_NETWORK_NAME}
+    sudo sed -i "/<\/dhcp>/i       <bootp file='boot.ipxe'/>" ${WORKING_DIR}/${BAREMETAL_NETWORK_NAME}
+    sudo sed -i "/<\/ip>/i     <tftp root='${PXE_SERVER_DIR}'/>" ${WORKING_DIR}/${BAREMETAL_NETWORK_NAME}
+
+    sudo virsh net-define ${WORKING_DIR}/${BAREMETAL_NETWORK_NAME}
+    sudo virsh net-destroy ${BAREMETAL_NETWORK_NAME}
+    sudo virsh net-start ${BAREMETAL_NETWORK_NAME}
+
+    # Copy the generated PXE artifacts in the tftp server location
```

Comment:
If there's a way we can use HTTP here that would be preferable. It's what our users will actually do, and will probably be more reliable for CI as well. The main reason for using iPXE is to avoid TFTP as much as possible.

Reply:
Ok, found a way that looks simple enough and use http, kept in a separate commit for better comparison. Local tests worked fine.

Reply:
Switch over also to direct use of dnsmasq options via xml, which seems to be a more promising approach to support in future also ipv6 (yet to be tested), but at least for the ipv4 case seems equal to the usage of `bootp` element

Reply:
So, it looks like that dnsmasq option approach worked well with ipv6, so added the support also for that